### PR TITLE
feat: pass file name to add/addAll progress handler

### PR DESCRIPTION
**Pull request by achingbrain**
Since ipfs/js-ipfs-unixfs#87 landed we can now pass the file name to
the progress handler for adding files:

```js
await ipfs.addAll(..., {
  progress: (bytes, fileName) => {
    //...
  }
})
```

This should make showing progress a bit more usable.

### CODE REVIEW

**Review by vasco-santos — APPROVED** [review]
LGTM!

File: packages/ipfs-core/src/components/add-all/index.js [hunk 1]
```diff
@@ -47,9 +47,9 @@ module.exports = ({ block, gcLock, preload, pin, options: constructorOptions })
       let total = 0
       const prog = opts.progress
 
-      opts.progress = (bytes) => {
+      opts.progress = (bytes, file) => {
```
Comment (hugomrdias) [inline packages/ipfs-core/src/components/add-all/index.js:50]:
```suggestion
      opts.progress = (bytes, fileName) => {
```

File: packages/ipfs-core/src/components/add-all/index.js [hunk 2]
```diff
@@ -47,9 +47,9 @@ module.exports = ({ block, gcLock, preload, pin, options: constructorOptions })
...
         total += bytes
-        prog(total)
+        prog(total, file)
```
Comment (hugomrdias) [inline packages/ipfs-core/src/components/add-all/index.js:52]:
```suggestion
        prog(total, fileName)
```

File: packages/ipfs-core/src/components/add-all/index.js (hunk 1 above)
Comment (hugomrdias) [reply packages/ipfs-core/src/components/add-all/index.js:50]: to match the jsdoc

**Review by hugomrdias — CHANGES_REQUESTED** [review]
Beside the inline suggestions LGTM

**Review by hugomrdias — APPROVED** [review]

**Pull request merged**